tests/mir-opt/dataflow-const-prop/default_boxed_slice.rs

```diff
@@ -11,8 +10,20 @@ struct A {
 
 // EMIT_MIR default_boxed_slice.main.GVN.diff
 // EMIT_MIR default_boxed_slice.main.DataflowConstProp.diff
+
+// CHECK-LABEL: fn main(
 fn main() {
     // ConstProp will create a constant of type `Box<[bool]>`.
     // Verify that `DataflowConstProp` does not ICE trying to dereference it directly.
+
+    // CHECK: debug a => [[a:_.*]];
+    // CHECK: scope {{[0-9]+}} (inlined <Box<[bool]> as Default>::default) {
+    // CHECK: scope {{[0-9]+}} (inlined Unique::<[bool; 0]>::dangling) {
+    // CHECK: scope {{[0-9]+}} (inlined NonNull::<[bool; 0]>::dangling) {
+    // We may check other inlined functions as well...
+
+    // CHECK: bb{{[0-9]+}}: {
+    // CHECK: [[box_obj:_.*]] = Box::<[bool]>(_3, const std::alloc::Global);
```

Comment:
```suggestion
    // CHECK: [[box_obj:_.*]] = const Box::<[bool]>
```
We want to verify that we have constants of that type, since those used to trigger the dataflow-const-prop bug.

Reply:
Let me dig into the history to understand the situation here first. However, the current MIR after constprop doesn't contain any patterns like "const Box::<[bool]>".

Reply:
I added a FIXME here.